back/spec/acceptance/baskets_ideas_spec.rb

```diff
@@ -0,0 +1,140 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+require 'rspec_api_documentation/dsl'
+
+resource BasketsIdea do
+  explanation 'Ideas included in a basket, with the count of votes.'
+
+  before do
+    header 'Content-Type', 'application/json'
+    @user = create(:user)
+    @project = create(:continuous_single_voting_project)
+    @basket = create(:basket, participation_context: @project, user: @user)
+  end
+
+  context 'when resident' do
+    before { header_token_for @user }
+
+    get 'web_api/v1/baskets/:basket_id/baskets_ideas' do
+      with_options scope: :page do
+        parameter :number, 'Page number'
+        parameter :size, 'Number of events per page'
+      end
+
+      let(:basket_id) { @basket.id }
+      let!(:ideas) { create_baskets_ideas @basket, votes: [3, 2, 1] }
+
+      example_request 'List all baskets_ideas of a basket' do
+        assert_status 200
+        json_response = json_parse response_body
+
+        expect(json_response[:data].pluck(:attributes).pluck(:votes)).to contain_exactly 3, 2, 1
+        expect(json_response[:included].select { |included| included[:type] == 'idea' }.pluck(:id)).to match_array ideas.map(&:id)
+      end
+    end
+
+    get 'web_api/v1/baskets_ideas/:id' do
+      let!(:ideas) { create_baskets_ideas @basket, votes: [3, 2, 1] }
+      let(:id) { @basket.baskets_ideas.where(votes: 2).first.id }
+
+      example_request 'Get one baskets_idea by ID' do
+        assert_status 200
+        json_response = json_parse response_body
+        expect(json_response.dig(:data, :id)).to eq id
+        expect(json_response.dig(:data, :attributes, :votes)).to eq 2
+        expect(json_response[:included].pluck(:id)).to include ideas[1].id
+      end
+    end
+
+    post 'web_api/v1/baskets/:basket_id/baskets_ideas' do
+      with_options scope: :baskets_idea do
+        parameter :idea_id, 'The ID of the idea added to the basket.', required: true
+        parameter :votes, 'The number of times the idea is voted on. Defaults to 1.', required: false
+      end
+      ValidationErrorHelper.new.error_fields self, BasketsIdea
+
+      let(:basket_id) { @basket.id }
+      let(:idea_id) { create(:idea, project: @project).id }
+      let(:votes) { 3 }
+
+      example_request 'Add an idea to a basket' do
+        assert_status 201
+        json_response = json_parse response_body
+
+        expect(json_response.dig(:data, :attributes, :votes)).to eq 3
+        expect(json_response[:included].pluck(:id)).to include idea_id
+      end
+
+      context 'when budgeting' do
```

Comment:
I don't think it's necessary to repeat the same acceptance tests for every voting method. Acceptance tests are very slow, and we have the voting_methods specs to test for the behaviour of the different voting methods. On the other hand, acceptance tests give a much stronger guarantee that the application is doing what it should. That's why I added a case for budgeting, because we need to make sure that the changes (votes set to budget) are truly persisted (save is called) in the controller. This is not something we could test in the voting_methods.

Reply:
Sure, that's fair enough. Probably nice to have, maybe just something for the future as we add more, maybe choose other voting methods unless they specifically need to be budgeting.

Reply:
I agree. Budgeting is a bit a special voting method, so it probably makes more sense to choose another voting method by default.